curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/650757918a9e97418dc3d71a.md

```diff
@@ -20,7 +20,7 @@ Congratulations! You have completed the platformer game project!
 You should add an `else if` clause to check is the player's `x` position is greater than or equal to the checkpoint's `x` position and less than or equal to the checkpoint's `x` position plus `40`.
 
 ```js
-assert.match(code, /if\s*\(\s*index\s*===\s*checkpoints\.length\s*-\s*1\s*\)\s*\{\s*isCheckpointCollisionDetectionActive\s*=\s*false;?\s*showCheckpointScreen\(("|'|`)You reached the final checkpoint!\1\);?\s*movePlayer\(\s*\1ArrowRight\1,\s*0,\s*false\);?\s*\}\s*else\s*if\s*\(\s*player\.position\.x\s*>=\s*checkpoint\.position\.x\s*&&\s*player\.position\.x\s*<=\s*checkpoint\.position\.x\s\+\s*40\s*\)\s*\{\s*/)
+assert.match(code, /if\s*\(\s*index\s*===\s*checkpoints\.length\s*-\s*1\s*\)\s*\{\s*isCheckpointCollisionDetectionActive\s*=\s*false;?\s*showCheckpointScreen\(("|'|`)You reached the final checkpoint!\1\);?\s*movePlayer\(\s*\1ArrowRight\1,\s*0,\s*false\);?\s*\}\s*else\s*if\s*\(\s*player\.position\.x\s*>=\s*checkpoint\.position\.x\s*&&\s*player\.position\.x\s*<=\s*checkpoint\.position\.x\s*\+\s*40\s*\)\s*\{\s*/)
```

Comment:
I see @RodrigoHLC this is your first time contributing to our repository. Thank you soo much for your contribution and hard work, i checked the issue and your solution seems good!. I also checked that the test code can be made shorter by removing the `if` statement. something like this 

```js
assert.match(code, /\s*else\s*if\s*\(\s*player\.position\.x\s*>=\s*checkpoint\.position\.x\s*&&\s*player\.position\.x\s*<=\s*checkpoint\.position\.x\s*\+\s*40\s*\)\s*\{\s*/)
```

there's 2 regex. make sure you address the solution to both of them. also, please check the merge conflicts and unit tests. Best of luck ! 👏

Reply:
Hey @RodrigoHLC ! 

It looks like you have an unresolved comment here.
Please review the changes and let us know if you have any questions, so it can be resolved 👍

Reply:
Hello, and sorry for the delay. I was in the middle of going through the first git courses on fCC hoping to get a better grasp on things (as Joy mentioned, this is my first time contributing to this repository). I don't really know how to do some of the things I've been requested to do, I just saw that there was something that needed fixing and tried to do it myself instead of just reporting the issue. I'm sorry if that's not enough, but I don't know what the "unit tests" are nor how to run this locally (I have now became familiar with Git, but that's about it). If you deem it preferable, feel free to ignore or cancel this contribution. Per my level of knowledge, this is as far as I can go right now ^^' Thanks, and have a good day everybody!

Reply:
Hi @RodrigoHLC ! 

> I don't know what the "unit tests" are nor how to run this locally 

These are the test run on each PR.
<img width="673" alt="Screenshot 2024-02-12 at 8 01 20 AM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/22429e46-9a75-4a15-908a-f2c3966d6385">

To run tests on a project locally, you can follow the directions from the documentation here
https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges?id=testing-challenges

As for the merge conflicts, here is a good article on how to do that locally
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

> Per my level of knowledge, this is as far as I can go right now ^^

It is up to you if you want to close this PR.
If you want to close it, then you can click on the "Close Pull Request" button at the bottom here
<img width="230" alt="Screenshot 2024-02-12 at 8 04 28 AM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/9af5350f-0a79-4859-8e09-062ba45c5e76">

But this could be a good learning opportunity for you if you wish to keep going with this PR.
The best way to learn Git is by practicing 👍 

Hope that helps